cpp/harvest/nacjd.cc

```diff
@@ -452,6 +444,170 @@ void convertJSONtoMARC(int argc, char **argv) {
     LOG_INFO("Finished.");
 }
 
+void getStatistics(int argc, char **argv) {
+    // Parse args
+    if (argc != 3)
+        Usage();
+
+    std::string json_path = argv[2];
+
+    std::string json_document;
+    FileUtil::ReadStringOrDie(json_path, &json_document);
+    JSON::Parser json_parser(json_document);
+    std::shared_ptr<JSON::JSONNode> internal_tree_root;
+    if (not json_parser.parse(&internal_tree_root))
+        LOG_ERROR("Could not properly parse \" " + json_path + ": " + json_parser.getErrorMessage());
+
+    const auto root_node(JSON::JSONNode::CastToObjectNodeOrDie("tree_root", internal_tree_root));
+    unsigned total(0), no_title(0), no_description(0), no_ID(0), no_alternate_ID(0), no_creators(0), no_license(0), no_initial_date(0),
+        no_keywords(0);
+
+    std::shared_ptr<JSON::ArrayNode> nacjd_nodes(JSON::JSONNode::CastToArrayNodeOrDie("nacjd", root_node->getNode("nacjd")));
+    for (const auto &internal_nacjd_node : *nacjd_nodes) {
+        ++total;
+
+        std::string title;
+        std::string description;
+        std::string node_ID;
+        std::string node_alternate_ID;
+        std::string creator;
+        std::string license;
+        std::string initial_release_date;
+        std::map<std::string, std::string> creators;
+        std::set<std::string> keywords;
+
+        const auto nacjd_node(JSON::JSONNode::CastToObjectNodeOrDie("entry", internal_nacjd_node));
+
+        const auto identifier_node(nacjd_node->getObjectNode("identifier"));
+        const auto alternate_identifiers_node(nacjd_node->getArrayNode("alternateIdentifiers"));
+        const auto creators_node(nacjd_node->getArrayNode("creators"));
+        const auto distributions_node(nacjd_node->getArrayNode("distributions"));
+        const auto keywords_node(nacjd_node->getOptionalArrayNode("keywords"));
+
+        const auto title_node(nacjd_node->getOptionalStringNode("title"));
+        if (title_node == nullptr) {
+            ++no_title;
+        } else {
+            title = title_node->getValue();
+            if (title.empty()) {
+                ++no_title;
+            }
+        }
+
+        const auto description_node(nacjd_node->getOptionalStringNode("description"));
+        if (description_node == nullptr) {
+            ++no_description;
+        } else {
+            description = description_node->getValue();
+            if (description.empty()) {
+                ++no_description;
+            }
+        }
+
+        if (keywords_node != nullptr) {
+            for (const auto &internal_keyword_node : *keywords_node) {
+                const auto keyword_node(JSON::JSONNode::CastToObjectNodeOrDie("keyword", internal_keyword_node));
+                const auto keyword_value_node(keyword_node->getStringNode("value"));
+                if (not keyword_value_node->getValue().empty())
+                    keywords.emplace(keyword_value_node->getValue());
+            }
```

Comment:
If the keywords set is empty after this loop we also want do perform `++no_keywords`.